lib/zwave/ZwaveDevice.js

```diff
@@ -283,9 +299,9 @@ class ZwaveDevice extends MeshDevice {
 		if (typeof parser !== 'function') return new Error('invalid_parser');
 
 		// Parse and check value
-		const parsedValue = parser.call(this, value, settingObj);
+		let parsedValue = parser.call(this, value, settingObj);
 		if (parsedValue instanceof Error) return parsedValue;
-		if (!Buffer.isBuffer(parsedValue)) return new Error('invalid_buffer');
+		if (!Buffer.isBuffer(parsedValue)) parsedValue = this._systemSettingParser(parsedValue, settingObj);
```

Comment:
What does this exactly do? A few lines above, the `_systemSettingParser()` is already called, so if it is still no `Buffer` afterwards it is not a good idea to try to parse it again.

Reply:
systemparser already makes sure if the value is a buffer, if not it just doesn't return it

so if the systemparser was used in parsedValue variable, then that call will never be triggered.

Reply:
Well, the thing is, in theory `if (!Buffer.isBuffer(parsedValue))` will never be entered, since `_systemSettingParser` only returns a `Buffer` or an `Error`, the if statement was only added in case some undefined error/bug occurred inside the `_systemSettingParser`, which could result in a return value other then `Buffer` or `Error`, in that case the `_parseSetting` method would either way return an `Error`. If I understand your change correctly, the only thing it does now is in the unlikely event the return value of the `_systemSettingParser` is not a `Buffer` or an `Error` it will try to parse it again. Which is strange, since you don't know the input value, but you do know that it failed to parse it the first time, so why parse it a second time?

Reply:
No, my change is for the normal parser (the custom one in the driver) like described in my pr this makes it so the value returned from the parser doesn't *need* to be a buffer, but can also be a number or boolean.
As other developers that don't know much about buffer can still write an app and not worry about for example size 2 and 4 number values.
It has nothing to do with the system parser before hand.

I know what you are getting at.
So might need to look into making sure it was this parser used instead of an end result and thus prevail a very small chance of a loop.